apps/portal-react/store/pathsEpic.tsx

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable prettier/prettier */
+import { filter, switchMap, from, tap } from 'rxjs';
+import { loadedPaths, requestPaths } from './paths';
+import { Action } from 'redux';
+import { Observable, map } from 'rxjs';
+import { db } from '../firebase/database';
+import { collection, getDocs } from 'firebase/firestore';
+
+export const loadPaths = (action$: Observable<Action>) => 
+    action$.pipe(
+        filter((action) => action.type === requestPaths.type),
+        switchMap(() =>
+            from(getDocs(collection(db, 'paths'))).pipe(
+                //tap((collection) =>  console.log(collection)),
+                map((collection) => loadedPaths(collection["_snapshot"].docChanges.map(path => {
+                    const fields = path.doc.data.value.mapValue.fields;
+                    const reqs = fields.requirements ? fields.requirements.arrayValue.values.map(e => e.referenceValue.match(/\w+$/i).toString()) : "none";
```

Comment:
The way this checks for the nullish-ness of `fields.requirements` is perfectly good. For fun, there is also a more terse way to write this:

```ts
const reqs = fields.requirements?.arrayValue.values.map(e => e.referenceValue.match(/\w+$/i).toString()) ?? "none";
```

where the `requirements?` checks if it is nullish, and if so, uses the value to the right of the "nullish coalescing operator" `??`, `"none"`